src/timesteppers.jl

```diff
@@ -480,3 +480,30 @@ function getetdcoeffs(dt, L; ncirc=32, rcirc=1)
 
   ζ, α, β, Γ
 end
+
+"""
+    step_until!(prob, stop_time)
+
+Step forward `prob` up to time `stop_time`.
+"""
+function step_until!(prob::Problem, stop_time::Union{Float64, Float32, Float16})
+  step_until!(prob::Problem, prob.timestepper::AbstractTimeStepper, stop_time)
+  return nothing
+end
+
+function step_until!(prob::Problem, ::Union{ETDRK4TimeStepper, FilteredETDRK4TimeStepper}, stop_time)
+  @error "step_until! requires fully explicit time stepper; does not work with ETDRK4"
+  return nothing
+end
+
+function step_until!(prob::Problem, timestepper, stop_time)
+  Δt = stop_time - prob.clock.t
```

Comment:
perhaps we should make sure that Δt>0?

Reply:
Check that `stop_time` is greater than `prob.clock.t`.

Reply:
How to do that without if?